### PR TITLE
SOLR-15177: Attempt to link to javadoc and refguide in htaccess

### PR DESCRIPTION
**Pull request by janhoy**
Non-tested rewrite rule magic to link in refguide at https://solr.apache.org/guide/X_Y and javadoc at https://solr.apache.org/docs/X_Y_Z/ on the new Solr site

https://issues.apache.org/jira/browse/SOLR-15177

### CODE REVIEW

File: themes/solr/templates/htaccess.template [hunk 1]
```diff
@@ -46,33 +46,32 @@ RedirectMatch Permanent ^mirrors-solr-3x-redir downloads.html
...
 # Hack: Append slash, if folder name does not end with slash - everything containing a "." is treated as file and excluded from redirect:
-RewriteRule ^(core|solr)/\d+_\d+_\d+/(?!.+\.\w+$|.+/$).+$   $0/ [R=301,L]
-RewriteRule ^solr/guide/\d+_\d+/(?!.+\.\w+$|.+/$).+$        $0/ [R=301,L]
+RewriteRule ^docs/\d+_\d+_\d+/(?!.+\.\w+$|.+/$).+$          $0/ [R=301,L]
+RewriteRule ^guide/\d+_\d+/(?!.+\.\w+$|.+/$).+$             $0/ [R=301,L]
 
 # __root/docs.lucene.apache.org/content/ is a special alias added by INFRA-19439, so we can refer to stuff in other places like SVN:
-RewriteRule ^(core|solr)/\d+_\d+_\d+/.*$  __root/docs.lucene.apache.org/content/$0 [PT]
-RewriteRule ^solr/guide/\d+_\d+/.*$       __root/docs.lucene.apache.org/content/$0 [PT]
+RewriteRule ^docs/(\d+_\d+_\d+/.*)$  __root/docs.lucene.apache.org/content/solr/$1 [PT]
```
Comment (uschindler) [inline themes/solr/templates/htaccess.template:75]: should work using capture group, I think!

File: themes/solr/templates/htaccess.template [hunk 2]
```diff
@@ -46,33 +46,32 @@ RedirectMatch Permanent ^mirrors-solr-3x-redir downloads.html
...
 
 # Solr Tutorial is now in the Solr Ref Guide
 # should redirect automatically to latest version
-RedirectMatch Permanent ^/quickstart.html /solr/guide/solr-tutorial.html
+RedirectMatch Permanent ^/quickstart.html /guide/solr-tutorial.html
 
 # redirect older point versions to latest (to keep number of javadocs small):
-RedirectMatch Permanent ^/solr/4_0_0-(ALPHA|BETA)(.*) /solr/4_0_0$2
-RedirectMatch Permanent ^/solr/api-(.*) /solr/$1
+RedirectMatch Permanent ^/docs/4_0_0-(ALPHA|BETA)(.*) /docs/4_0_0$2
```
Comment (uschindler) [inline themes/solr/templates/htaccess.template:65]: I am not sure if we need those, as the solr.apache.org site is new, so the "old" redirects should not be there anyways. It depends on the redirects on Lucene side: If we have a general perm (301) redirect from lucene.apache.org/solr/ -> solr.apache.org/, we should keep them here.

**Review by uschindler — APPROVED** [review]
Hi,
in general this should all work, see inlined comments.
Testing is hard, let's go to production and fix the links after release. No way otherwise.
But it's not too risky.

For the subversion folder we should look for a better alternative anywayss. Maybe also split the folder in svn. But for now it's OK.

File: themes/solr/templates/htaccess.template (hunk 1 above)
Comment (janhoy) [reply themes/solr/templates/htaccess.template:75]: Yes, looks like it is supported: https://httpd.apache.org/docs/2.4/rewrite/intro.html#InternalBackRefs

File: themes/solr/templates/htaccess.template (hunk 2 above)
Comment (janhoy) [reply themes/solr/templates/htaccess.template:65]: Yes, on the Lucene side of things I think we can do like this, except the /solr/X_Y_Z which must be rewritten to /docs/X_Y_Z. So putting those rules first and fallback to the generic rule shold do the trick.

**Pull request merged**